gulp-tasks/publish-glitch.js

```diff
@@ -0,0 +1,62 @@
+/*
+  Copyright 2018 Google LLC
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+const gulp = require('gulp');
+// const del = require('del');
+const git = require('gulp-git');
+const axios = require('axios');
+
+const key = '9c5af840-16dd-42d8-b5d1-0fa4a52939b8';
+
+const glitchProjects = [
+  'workbox-core',
+  // 'workbox-sw',
+  // 'workbox-precaching',
+  // 'workbox-strategies',
+  // 'workbox-background-sync-demo',
+  // 'workbox-routing',
+  // 'workbox-expiration',
+  // 'workbox-cacheable-response',
+  // 'workbox-google-analytics',
+  // 'workbox-streams',
+  // 'workbox-range-requests',
+  // 'workbox-broadcast-update-demo',
+  // 'workbox-window',
+  // 'workbox-navigation-preload'
+];
+
+gulp.task('publish-glitch', async () => {
+  for (const project of glitchProjects) {
+    // repo URLs can be associated w/ the demo folder through a predefined object OR th
+    // for each repo...
+    const projectURL = 'https://' + key + '@api.glitch.com/git/' + project;
+    const projectPath = './temp_' + project;
+    const date = new Date();
+
+    await git.clone(projectURL, {args: projectPath}, async () => {
```

Comment:
Assuming you have the promisified `exec` set up based on the earlier comment, you could move from `gulp-git` to

```
const tempDir = await tempy.directory();
await exec(`git clone ${projectURL}`, {cwd: tempDir});
// See https://github.com/jprichardson/node-fs-extra/blob/master/docs/copy.md
await fse.copy(..., ...);
// Etc.
```

Reply:
got it, makes sense. Given that I added axios as well, do you recommend using exec to invoke cURL for the hook request?

Reply:
I'm cool with using `axios` instead of shelling out to `cURL`. I view using `exec` as more of a second-choice when the `node`-native choice (like `gulp-git`) is way too much effort.

Reply:
(Or if you already have the code working with `cURL`, as you seem to, that's fine with me.)

Reply:
it should be good to go now. we can review on Tuesday if you'd like. I also have a write up on the full funnel.